### PR TITLE
Update some build and test dependencies | Update to latest design-system 0.37.x

### PR DESCRIPTION
**Pull request by anselmbradford**
Design system changes in https://github.com/cfpb/design-system/compare/v0.27.0...v0.37.2

Note that `a-micro-copy` has been removed from the design system, but this project sets its own styles on that class.

## Changes

- Update some build and test dependencies
- Update to latest design-system

## Testing

- PR checks should pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,35 +37,35 @@
...
+    "@babel/cli": "^7.23.9",
+    "@babel/core": "^7.23.9",
+    "@babel/eslint-parser": "^7.23.10",
+    "@babel/preset-react": "^7.23.3",
+    "@babel/runtime": "^7.23.9",
+    "@cfpb/cfpb-atomic-component": "0.37.0",
+    "@cfpb/cfpb-buttons": "0.37.0",
+    "@cfpb/cfpb-core": "0.37.0",
+    "@cfpb/cfpb-design-system": "^0.37.2",
+    "@cfpb/cfpb-expandables": "^0.37.0",
```
Comment (flacoman91) [inline package.json:49]: it all looks the same, but is this issue still valid?  

https://github.com/cfpb/ccdb5-ui/issues/463

File: package.json [hunk 2]
```diff
@@ -74,29 +74,29 @@
     "eslint": "^8.56.0",
```
Comment (flacoman91) [inline package.json:74]: looks like we haven't used `enzyme` and `enzyme-adapter-react-16` for a very long time too.  can we remove those two?

File: package.json (hunk 1 above)
Comment (anselmbradford) [reply package.json:49]: Updated here. Hopefully I got everything https://github.com/cfpb/ccdb5-ui/pull/490/commits/96b01034f59d78277900dc11c19c0fb4f66da5e0

Before:
Screenshot 2024-02-22 at 4 50 37 PM

After:
Screenshot 2024-02-22 at 4 50 47 PM

Comment (flacoman91) [reply package.json:49]: wow, that was fast.  Faster than me trying to figure it out!

**Review by flacoman91 — APPROVED** [review]
LGTM

**Pull request merged**